t/lib/OpenQA/Test/Utils.pm

```diff
@@ -172,11 +173,15 @@ sub redirect_output {
 }
 
 sub stop_service {
-    my ($pid, $forced) = @_;
-    return unless $pid;
-    kill POSIX::SIGTERM => $pid;
-    kill POSIX::SIGKILL => $pid if $forced;
-    waitpid($pid, 0);
+    my ($h, $forced) = @_;
+    return unless $h;
+    if ($forced) {
+        $h->kill_kill(grace => 3);
```

Comment:
Should we handle the exception and print a warning? Otherwise it presumably breaks the test plan in that case.

Reply:
I don't think it does. This is basically still the same we were doing in before, we send TERM and then KILL if the parameter was given